tests/data_context/test_data_context_utils.py

```diff
@@ -286,6 +287,222 @@ def test_password_masker_mask_db_url(monkeypatch, tmp_path):
+
+    # test that cloud store backend tokens have been properly masked
+    config_with_creds_in_stores = PasswordMasker.sanitize_config(
+        data_context_config_dict_with_cloud_backed_stores
+    )
+    for name, store_config in config_with_creds_in_stores["stores"].items():
+        try:
+            # check that the original token exists
+            assert (
+                data_context_config_dict_with_cloud_backed_stores["stores"][name][
+                    "store_backend"
+                ]["ge_cloud_credentials"]["access_token"]
+                == ge_cloud_access_token
+            )
+            # expect that the GE Cloud token has been obscured
+            assert (
+                store_config["store_backend"]["ge_cloud_credentials"]["access_token"]
+                != ge_cloud_access_token
+            )
+        except KeyError:
+            # a field in store_config["store_backend"]["ge_cloud_credentials"]["access_token"]
+            # doesn't exist, so we expect this config to be unchanged
+            assert (
+                store_config
+                == data_context_config_dict_with_cloud_backed_stores["stores"][name]
+            )
+
+
+def test_sanitize_config_masks_execution_engine_connection_strings(
+    data_context_config_dict_with_datasources, conn_string_password
+):
+    """Test case for PasswordMasker.sanitize_config"""
+
+    # test that datasource credentials have been properly masked
+    unaltered_datasources = data_context_config_dict_with_datasources["datasources"]
+    config_with_creds_masked = PasswordMasker.sanitize_config(
+        data_context_config_dict_with_datasources
+    )
+    masked_datasources = config_with_creds_masked["datasources"]
+
+    # iterate through the processed datasources and check for correctness
+    for name, processed_config in masked_datasources.items():
+
+        # check if processed_config["execution_engine"]["connection_string"] exists
+        if processed_config.get("execution_engine") and processed_config[
+            "execution_engine"
+        ].get("connection_string"):
+
+            # check if the connection string contains a password
+            if (
+                conn_string_password
+                in unaltered_datasources[name]["execution_engine"]["connection_string"]
+            ):
+                # it does contain a password, so make sure its masked
+                assert (
+                    conn_string_password
+                    not in processed_config["execution_engine"]["connection_string"]
+                )
+            else:
+                # it doesn't contain a password, so make sure it's unaltered
+                assert processed_config == unaltered_datasources[name]
+
+        # processed_config either doesn't have an `execution_engine` field,
+        # or a `connection_string` field
+        else:
+            # expect this config to be unaltered
+            assert processed_config == unaltered_datasources[name]
+
```

Comment:
Same general thought here! Could we just assert against a dict so we know the full expected vs actual?